src/components/QuestionAnswer.js

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect, useState } from 'react';
+import { subscribe } from './Event';
+import { getQuestion } from './Helper';
+
+function QuestionAnswer(props) {
+  const { questionNum, nextQuestion, quiz } = props;
+  const [questionObj, setQuestion] = useState(getQuestion(questionNum, quiz));
+  const [inputVal, setInputVal] = useState('');
+  useEffect(() => {
+    subscribe(`NextQuestion${quiz.quizId}`, () => {
+      nextQuestion(questionObj);
+      setQuestion(getQuestion(questionNum + 1, quiz));
+      setInputVal('');
+    });
+  });
+  const handleOnClick = () => {
+    nextQuestion(questionObj);
+    setQuestion(getQuestion(questionNum + 1, quiz));
+    setInputVal('');
+  };
+  const onChangeHandler = (e) => {
+    setQuestion((prevState) => ({
```

Comment:
Do we need prevState here? When do we need prevState?

Reply:
Done